### ISSUE TITLE
Clean up use of unittest assert functions in twisted/runner

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#8390]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)| @rodrigc reported|
|-|-|
|Trac ID|trac#8390|
|Type|enhancement|
|Created|2016-06-03 02:14:24Z|

This ticket is split out from [ticket:8371] because the patch there was too big to review.

The files in twisted/runner
should have their use of unittest assert functions cleaned up so that:

* the most appropriate unittest assert function is used appropriately in each file
* use of deprecated aliases should be eliminated, to reduce deprecation
  warnings and errors when moving to Python3 compatibility

This has some pointers about deprecated unittest assert aliases:

See: https://docs.python.org/2.7/library/unittest.html#deprecated-aliases

|= Method Name =|= Deprecated alias(es) =|
|---------------|------------------------|
| assertEqual() | failUnlessEqual, assertEquals |
| assertNotEqual() | failIfEqual |
| assertTrue() | failUnless, assert_ |
| assertFalse() | failIf |
| assertRaises() |	failUnlessRaises |
| assertAlmostEqual() | failUnlessAlmostEqual  |
| assertNotAlmostEqual() | failIfAlmostEqual  |

<details><summary>Searchable metadata</summary>

```
trac-id__8390 8390
type__enhancement enhancement
reporter__rodrigc rodrigc
priority__normal normal
milestone__None None
branch__ 
branch_author__ 
status__closed closed
resolution__fixed fixed
component__core core
keywords__None None
time__1464920064969271 1464920064969271
changetime__1465876435831919 1465876435831919
version__None None
owner__glyph glyph

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_1"></a>|@rodrigc commented|
|-|-|

See:
https://github.com/twisted/twisted/pull/106

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_3"></a>|@glyph set owner to @rodrigc|
|-|-|

Thanks for your contribution, rodrigc.

As with [#8376](https://github.com/twisted/twisted/issues/8376), the changes to these assertion functions are over-broad.  These are a lot of changes to look over, but please re-submit more minimal changes that just address the necessary bits to migrate to python 3.

**Comment by twisted-trac** [comment]
|[<img alt="rodrigc's avatar" src="https://avatars.githubusercontent.com/u/1895943?s=50" width="50" height="50">](https://github.com/rodrigc)<a name="note_4"></a>|@rodrigc set owner to @glyph|
|-|-|

glyph, I understand you were busy at Pycon, and I do agree that some of the reviews I submitted were large.
I have been working on creating smaller patches and sending them for review.

However, for this review, the patch I submitted changed 3 lines.

I don't see how this is an "over-broad" change.  It looks like you cut and paste your response to multiple of my reviews
without even looking at the patch.  I don't agree with this review being over-broad.

Please look at the patch, and provide more detailed feedback.

Thanks.

**Comment by twisted-trac** [comment]
|[<img alt="glyph's avatar" src="https://avatars.githubusercontent.com/u/716529?s=50" width="50" height="50">](https://github.com/glyph)<a name="note_5"></a>|@glyph commented|
|-|-|

Replying to [rodrigc](#note_4):
> glyph, I understand you were busy at Pycon, and I do agree that some of the reviews I submitted were large.
> I have been working on creating smaller patches and sending them for review.
> 
> However, for this review, the patch I submitted changed 3 lines.
> 
> I don't see how this is an "over-broad" change.  It looks like you cut and paste your response to multiple of my reviews
> without even looking at the patch.  I don't agree with this review being over-broad.

I did copy/paste (as you clearly did with your descriptions) but not without looking at the diffs.  I scanned each diff based on the table that you put in the description, to see which deprecated aliases were being touched, as opposed to other, less functional renamings.  In the case of this branch, based on what I was evaluating for at the time, I would stand by that review, since neither of `assertIdentical` or `assertIn` are deprecated aliases in python 2.

I should probably have put in a more explanatory message: these tickets were all classified as part of the python 3 porting effort, _and they are unnecessary for porting to python 3_.  Some of the changes, as I explained on [#8376](https://github.com/twisted/twisted/issues/8376), were actually neutral or maybe even a little counterproductive, such as changing `assertEqual(True, whatever)` to `assertTrue(whatever)`.

However, in this case, upon further reflection, I think the patch _is_ useful - for other reasons.  The `assertIn` change is an improvement to clarity and the usefulness of the error message when there's a failure; and it appears that, independent of python 3 porting, `assertIdentical` appears to be an [entirely undocumented](http://twistedmatrix.com/documents/16.2.0/api/twisted.trial.unittest.TestCase.html) alias that we made up before the existence of the new-in-2.7 `assertIs` in the standard library.  So we should probably go ahead and merge this anyway.

> Please look at the patch, and provide more detailed feedback.

This one does look fine, for the reasons given above; thanks for reopening it.

**Comment by twisted-trac** [comment]
|<img alt="GitHub's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|GitHub set status to `closed`|
|-|-|

In changeset 1a0e1d1adcaa3913704b9868d80d1a15bb41dfcf
```
#!CommitTicketReference repository="" revision="1a0e1d1adcaa3913704b9868d80d1a15bb41dfcf"
Merge pull request #106 from rodrigc/8390-rodrigc-assert

Author: rodrigc

Reviewer: glyph

Fixes: #8390

Clean up a few assertion methods in twisted.runner's test suite:

- `assertIsNone` rather than our undocumented `assertIdentical`
- more expressive `assertIn` rather than `assertTrue`
```
